src/ripple/app/main/Application.cpp

```diff
@@ -1976,6 +1976,17 @@ bool ApplicationImp::loadOldLedger (
                 }
             }
         }
+        using namespace std::chrono_literals;
+        static constexpr char ledgerWarnTime[] {"Jan 01, 2018 00:00:00 UTC"};
+        static constexpr NetClock::time_point ledgerWarnTimePoint {568080000s};
```

Comment:
Boy, I wish I had a better answer for this question.  And I will once we're building with C++20.  In C++20 `chrono` provides calendar support.  Once we have calendar support then the string can be generated from the `time_point`.  In the meantime I think this is the best we can do.  But possibly @HowardHinnant has a better approach?

Reply:
Here's a tested alternative: https://github.com/HowardHinnant/rippled/commit/3b1a9b7acbea9afc7a8d5a76a3bb2ee7fdc4c878

It isn't great.  I think it could be better but would involve a change to the draft C++20 spec (which this has motivated me to try for).

Reply:
This is a minor nit really; no change is necessary to this for me. But looking forward to what Howard comes up with ;)

Reply:
We can do `to_string(ledgerWarnTimePoint)` today.  The output will be `2018-Jan-01 00:00:00 UTC` instead of `Jan 01, 2018 00:00:00 UTC`, which I think is insignificant.  That gets rid of `ledgerWarnTime`.

Now the only question is how to best construct `ledgerWarnTimePoint`:

Status quo:
```
constexpr NetClock::time_point ledgerWarnTimePoint {568080000s};
```
Proposal we can do today:
```
constexpr NetClock::time_point ledgerWarnTimePoint{
                          sys_days{January/1/2018} - sys_days{January/1/2000}};
```

In both of the above cases `static` is unnecessary.

What I believe I'm going to try to do:

```
constexpr auto ledgerWarnTimePoint = clock_cast<NetClock>(sys_days{January/1/2018});
```
We can almost do this today, except we can't make it `constexpr`.  Making it `constexpr` is what I think I want to try to do.  I've already got a PR from someone requesting this change.